x-pack/metricbeat/module/oracle/sysmetric/data.go

```diff
@@ -0,0 +1,43 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package sysmetric
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/beats/v7/metricbeat/mb"
+)
+
+// extract is the E of a ETL processing. Gets all the system metric information into an instance of extractedData
+func (m *MetricSet) extract(ctx context.Context, extractor sysmetricExtractMethod) (out *extractedData, err error) {
```

Comment:
From the definition of ETL, it stands for extract, transform and load, is a data integration process that combines data from multiple data sources into a single, consistent data store that is loaded into a  datawarehouse. It is a term commonly used in datawarehousing domain not in observability 

I understand that similar function names is used for performance and tablespace. 

I do not not think that it is a correct name to be followed for an observability product. 



Recommendation : Suggest to use different function names for extract, extranctAndTransform and transform. Also, avoid comments that says that this is a ETL specific operation that is happening.